src/Persistence/Util/ClassUtils.php

```diff
@@ -0,0 +1,106 @@
+<?php
+
+namespace Doctrine\Persistence\Util;
+
+use Doctrine\Persistence\Proxy;
+use ReflectionClass;
+
+/**
+ * Class and reflection related functionality for objects that
+ * might or not be proxy objects at the moment.
+ */
+class ClassUtils
```

Comment:
`ProxyConverter`? There has to be a better name…

Reply:
ProxyResolver ?